File: Trimestre 6/Proyecto final/APP MOVIL/API REST/src/models/detalleVenta.model.js
import { Model, DataTypes } from "sequelize";
import sequelize from "../config/connect.db.js";
import Venta from "./venta.model.js";
import Producto from "./producto.model.js";

class DetalleVenta extends Model {}

DetalleVenta.init(
  {
    idDetalleVenta: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true,
    },
    Cantidad: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
    PrecioUnitario: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: false,
    },
    Subtotal: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: false,
    },
    Venta_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: Venta,
        key: "idVenta",
      },
    },
    Producto_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: Producto,
        key: "idProducto",
      },
    },
  },
  {
    sequelize,
    modelName: "DetalleVenta",
    tableName: "DetalleVenta",
    timestamps: false,
  }
);

export default DetalleVenta;

File: Trimestre 6/Proyecto final/Proyecto final/MOVIL/appmovil/lib/models/detalle_venta_model.dart
class DetalleVentaModel {
  final int idDetalleVenta;
  final int ventaId;
  final int productoId;
  final int cantidad;
  final double precioUnitario;
  final double subtotal;

  DetalleVentaModel({
    required this.idDetalleVenta,
    required this.ventaId,
    required this.productoId,
    required this.cantidad,
    required this.precioUnitario,
    required this.subtotal,
  });

  factory DetalleVentaModel.fromJson(Map<String, dynamic> json) {
    return DetalleVentaModel(
      idDetalleVenta: int.tryParse((json['idDetalleVenta'] ?? json['id']).toString()) ?? 0,
      ventaId: int.tryParse((json['Venta_id'] ?? json['ventaId']).toString()) ?? 0,
      productoId: int.tryParse((json['Producto_id'] ?? json['productoId']).toString()) ?? 0,
      cantidad: int.tryParse((json['Cantidad'] ?? json['cantidad']).toString()) ?? 0,
      precioUnitario: double.tryParse((json['PrecioUnitario'] ?? json['precioUnitario'] ?? 0).toString()) ?? 0,
      subtotal: double.tryParse((json['Subtotal'] ?? json['subtotal'] ?? 0).toString()) ?? 0,
    );
  }

  Map<String, dynamic> toJson() => {
        'Cantidad': cantidad,
        'PrecioUnitario': precioUnitario,
        'Subtotal': subtotal,
        'Venta_id': ventaId,
        'Producto_id': productoId,
      };
}
